app/common/data/interfaces/user.py

```diff
@@ -29,6 +29,13 @@ def get_platform_admin_users() -> Sequence[User]:
     ).all()
 
 
+def get_user_by_email(email_address: str) -> Optional[User]:
+    try:
+        return db.session.execute(select(User).where(User.email == email_address)).scalar_one()
+    except NoResultFound:
+        return None
```

Comment:
totally optional but I think this can more succinctly be written as this (I've not tried this so if doing please do check it passes tests and does what you'd expect! note: neither of these handle the case where there's for some reason too many users for the email which I think is valid as the DB should constrain that in theory)

```suggestion
return db.session.execute(select(User).where(User.email == email_address)).scalar_one_or_none()```

Reply:
it works & I did not know that they have this in place thanks @sfount